config/filter.d/exim-common.conf

```diff
@@ -9,7 +9,7 @@ after = exim-common.local
 
 [Definition]
 
-host_info = H=([\w.-]+ )?(\(\S+\) )?\[<HOST>\](:\d+)? (I=\[\S+\]:\d+ )?(U=\S+ )?(P=e?smtp )?
+host_info = (H=([\w.-]+ )?(\(\S+\) )?)?\[<HOST>\](:\d+)? (I=\[\S+\](:\d+)? )?(U=\S+ )?(P=e?smtp )?
```

Comment:
I don't like usage of capturing groups, so think `(?:)` would be generally better here :))